### PR TITLE
a possible fix for the regression aaronFranke found

### PR DESCRIPTION
**Pull request by Lcbx**
might be a fix for aronFranke's concerns :
use the name of the file when the scene name is not set or begins with 'Scene'
then use the scene name when naming the mesh

I also reused `_gen_unique_name` and removed the whitespace in the generated string

It seems to do ok on the TPS demo. The names are longer but there should be no collisions.

### CODE REVIEW

**Comment by fire** [comment]
```c++
String mesh_name = "mesh";
if (d.has("name" && d["name"] != "")) {
	mesh_name = d["name"];
}
```

This logic is a bit weird.

**Comment by fire** [comment]
Please try https://github.com/godotengine/godot/pull/46783.

**Comment by Lcbx** [comment]
> ```c++
> String mesh_name = "mesh";
> if (d.has("name" && d["name"] != "")) {
> 	mesh_name = d["name"];
> }
> ```
> 
> This logic is a bit weird.

yeah maybe this would make more sense : 
```c++
String mesh_name = d.has("name") && !String(d["name"]).empty() ? d["name"] : "mesh";
 ```

**Comment by Lcbx** [comment]
Hello

I tried to make a new PR for the new commit (that restores original behaviour when there's only one mesh)
but it seems like you can only have one PR from one branch to another at the same time

**Comment by fire** [comment]
Maybe drop a patch.

**Comment by Lcbx** [comment]
not sure how to do that

File: modules/gltf/gltf_document.cpp [hunk 1]
```diff
@@ -2534,11 +2534,12 @@ Error GLTFDocument::_parse_meshes(Ref<GLTFState> state) {
...
 		import_mesh.instance();
-		String mesh_name = "mesh";
-		if (d.has("name") && !String(d["name"]).empty()) {
-			mesh_name = d["name"];
+		if (meshes.size() == 1) {
+			import_mesh->set_name(state->scene_name);
+		} else {
+			String mesh_name = (d.has("name") && !String(d["name"]).empty()) ? d["name"] : "mesh";
+			import_mesh->set_name(_gen_unique_name(state, vformat("%s_%s", state->scene_name, mesh_name)));
 		}
```
Comment (Lcbx) [inline modules/gltf/gltf_document.cpp:2542]: here's the new code.
now `BlastMesh.glb` becomes `BlastMesh.mesh` and not `BlastMesh_mesh.mesh`

Comment (fire) [reply modules/gltf/gltf_document.cpp:2542]: If it is saved to file I think godot uses `blastmesh_mesh.mesh` as the convention.

**Comment by fire** [comment]
Godot Engine has changed. Will close. Feel free to comment.

**Pull request closed without merging**